obs-studio-client/source/input.cpp

```diff
@@ -908,6 +910,30 @@ Nan::NAN_METHOD_RETURN_TYPE osn::Input::SetFilterOrder(Nan::NAN_METHOD_ARGS_TYPE
 
 	SourceDataInfo* sdi = CacheManager<SourceDataInfo*>::getInstance().Retrieve(baseobj->sourceId);
 	if (sdi) {
+		// Update order on cache manager filter container
+		auto it = std::find(sdi->filters->begin(), sdi->filters->end(), basefilter->sourceId);
+
+		if (movement == FilterMovement::UP && it != sdi->filters->begin()) {
+			std::iter_swap(it, it - 1);
+		}
```

Comment:
This 4 if's can be joined by "else" or replaced by "switch" as they expected to work exclusively. If one of them had worked no need to check another.

Reply:
You are right. I'll change it to a switch